src/Mobs/Path.cpp

```diff
@@ -113,15 +109,24 @@ ePathFinderStatus cPath::Step()
 
 
 
-#ifndef COMPILING_PATHFIND_DEBUGGER
 bool cPath::IsSolid(const Vector3d & a_Location)
 {
-	int ChunkX, ChunkZ;
-	m_Item_CurrentBlock = a_Location;
-	cChunkDef::BlockToChunk(a_Location.x, a_Location.z, ChunkX, ChunkZ);
-	return !m_World->DoWithChunk(ChunkX, ChunkZ, * this);
+	m_Chunk = m_Chunk->GetNeighborChunk(a_Location.x, a_Location.z);
+	if (!m_Chunk->IsValid())
+	{
+		return true;
+	}
+	BLOCKTYPE BlockType;
+	NIBBLETYPE BlockMeta;
+	int RelX = a_Location.x - m_Chunk->GetPosX() * cChunkDef::Width;
+	int RelZ = a_Location.z - m_Chunk->GetPosZ() * cChunkDef::Width;
+	m_Chunk->GetBlockTypeMeta(RelX, a_Location.y, RelZ, BlockType, BlockMeta);
+	if ((BlockType == E_BLOCK_FENCE) || (BlockType == E_BLOCK_FENCE_GATE))
```

Comment:
A closed fence-gate has a different meta than an open one.

As for the fences, consider the following situation:
![2015-05-02_09 55 30](https://cloud.githubusercontent.com/assets/4388386/7440385/97db8ac4-f0b1-11e4-9b6f-731c14a710ca.jpg)
Your code would allow the zombie to pathfind through to get me, simply because the cell on top of the fence is processed earlier than the cell with the fence.